packages/blocto-sdk/src/providers/ethereum.ts

```
@@ -52,114 +41,127 @@ export default class EthereumProvider
+    switchableNetwork: SwitchableNetwork;
+  };
 
   constructor({
     chainId,
     rpc,
-    server,
+    walletServer,
     appId,
     session,
   }: EthereumProviderConfig) {
     super(session);
+    // setup chainId
     invariant(chainId, "'chainId' is required");
-
-    this.chainId = parseChainId(chainId);
-    this.networkId = this.chainId;
-    this.chain = ETH_CHAIN_ID_CHAIN_MAPPING[this.chainId];
-    this.net = ETH_CHAIN_ID_NET_MAPPING[this.chainId];
-
-    invariant(this.chain, `unsupported 'chainId': ${this.chainId}`);
-
-    this.rpc = rpc || ETH_CHAIN_ID_RPC_MAPPING[this.chainId] || '';
-
-    invariant(this.rpc, "'rpc' is required for Ethereum");
-
-    this.server = server || ETH_CHAIN_ID_SERVER_MAPPING[this.chainId] || '';
+    this.networkVersion = `${parseChainId(chainId)}`;
+    this.chainId = `0x${parseChainId(chainId).toString(16)}`;
+    // setup rpc
+    this.rpc = rpc || ETH_RPC_LIST[this.networkVersion];
+    invariant(rpc, "'rpc' is required");
+    // setup injectedWalletServer
+    this.injectedWalletServer = walletServer;
+    // NOTE: _blocto is not fully initialized yet at this point
+    // Any function should call #getBloctoProperties() to get the full _blocto properties
+    this._blocto = {
+      session,
+      walletServer: this.injectedWalletServer || '',
+      blockchainName: '',
+      networkType: '',
+      supportNetworkList: {},
+      switchableNetwork: {},
+    };
     this.appId = appId || DEFAULT_APP_ID;
+  }
 
-    this.switchableNetwork[this.chainId] = {
-      name: this.chain,
-      display_name: this.chain,
-      network_type: this.net,
-      wallet_web_url: this.server,
-      rpc_url: this.rpc,
+  async #getBloctoProperties(): Promise<EthereumProvider['_blocto']> {
+    if (!Object.keys(this._blocto.supportNetworkList).length) {
+      await getEvmSupport()
+        .then((result) => (this._blocto.supportNetworkList = result))
+        .catch((e) => {
+          console.error(e);
```

It's better to throw this error out?

---

Ok, thanks.